pkg/crc/oc/oc.go

```diff
@@ -45,21 +47,25 @@ func UseOCWithConfig(machineName string) OcConfig {
 		OcBinaryPath:   filepath.Join(constants.CrcBinDir, constants.OcBinaryName),
 		KubeconfigPath: filepath.Join(constants.MachineInstanceDir, machineName, "kubeconfig"),
 	}
-	return NewOcConfig(localRunner)
+	return NewOcConfig(localRunner, constants.DefaultContext, constants.DefaultName)
 }
 
 func (oc OcConfig) RunOcCommand(args ...string) (string, string, error) {
-	args = append(args, "--kubeconfig", oc.runner.GetKubeconfigPath())
+	args = append(args, "--kubeconfig", oc.runner.GetKubeconfigPath(), "--context", oc.Context, "--cluster", oc.Cluster)
 	return oc.runner.Run(args...)
 }
 
 func (oc OcConfig) RunOcCommandPrivate(args ...string) (string, string, error) {
-	args = append(args, "--kubeconfig", oc.runner.GetKubeconfigPath())
+	args = append(args, "--kubeconfig", oc.runner.GetKubeconfigPath(), "--context", oc.Context, "--cluster", oc.Cluster)
 	return oc.runner.RunPrivate(args...)
 }
 
-func NewOcConfig(runner OcRunner) OcConfig {
-	return OcConfig{runner: runner}
+func NewOcConfig(runner OcRunner, context string, clusterName string) OcConfig {
```

Comment:
Is `context` something only valid in the context of `clusterName`? Ie should we have `clusterName` first in the argument list, and then `context`?

Reply:
@cfergeau yes both are independent options in the `oc options`

```
10:52 $ oc options 2>&1 | grep cluster
      --cluster='': The name of the kubeconfig cluster to use
10:52 $ oc options 2>&1 | grep context
      --context='': The name of the kubeconfig context to use
```